File: javascript/space-age/space-age.js
"use strict";

const to_earth_years = {
    "earth"   : 1,
    "mercury" : 0.2408467,
    "venus"   : 0.61519726,
    "mars"    : 1.8808158,
    "jupiter" : 11.862615,
    "saturn"  : 29.447498,
    "uranus"  : 84.016846,
    "neptune" : 164.79132
};

export const age = (planet, seconds) => Number(
    ((seconds / 31557600) / to_earth_years[planet]).toFixed(2)
);

File: haskell/space-age/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

type Seconds = Float

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

secsInEarthYear :: Seconds
secsInEarthYear = 60 * 60 * 24 * 365.25

ageOn :: Planet -> Float -> Float
ageOn planet seconds   = earth_years / in_earth_years
  where earth_years    = (seconds / secsInEarthYear)
        in_earth_years = case planet of
                           Earth   -> 1
                           Mercury -> 0.2408467
                           Venus   -> 0.61519726
                           Mars    -> 1.8808158
                           Jupiter -> 11.862615
                           Saturn  -> 29.447498
                           Uranus  -> 84.016846
                           Neptune -> 164.79132
